### PR TITLE
Import OptionError in pandas.errors

### LINKED ISSUE #27553: `OptionError` raised by top-level function `pandas.set_option` is private
#### Code Sample

```python
import pandas as pd
from pandas.errors import OptionError 
# or from pandas import OptionError
option = ('noexist', 1)
try:
    pd.set_option(*option)
except OptionError as e:
    print('Ignore invaild option: {}'.format(option))
```
#### Problem description

`OptionError` is raised by top-level function `pandas.set_option`, but can only be imported from private module `pandas._config.config`.

Sample code will failed when importing it from `pandas.errors`  :

```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-62b92256ce5f> in <module>
      1 import pandas as pd
----> 2 from pandas.errors import OptionError # or from pandas import OptionError
      3 option = ('noexist', 1)
      4 try:
      5     pd.set_option(*option)

ImportError: cannot import name 'OptionError' from 'pandas.errors' (/home/abc/.local/share/virtualenvs/abc-jIQt2SYy/lib/python3.7/site-packages/pandas/errors/__init__.py)
```

#### Expected Output

```
Ignored invaild option: ('noexist', 1)
```

Expected behavior : `OptionError` can be imported from `pandas.errors` or `pandas`. And other custom exceptions raised by top-level functions are also available in the same public namespace.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.4.0-17763-Microsoft
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.0
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 41.0.1
Cython           : 0.29.12
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.6.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : 2.6.2
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by sumanau7**
- [x] closes #27553
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by sumanau7** [comment]
@jorisvandenbossche I am experiencing a build failure on my forked master as well, is this some setup issue, would need your help to resolve this.

Also Is the pr good to go or more changes would be required ?

**Comment by jbrockmendel** [comment]
Travis failure should be unrelated, just restarted that build.

**Comment by jbrockmendel** [comment]
Can you add this to test_exception_importable in tests.test_errors

**Comment by sumanau7** [comment]
@jbrockmendel Tests are changed, a whatsnew entry is created. Would you be able to review this or suggest someone who can help in reviewing this PR.

File: doc/source/whatsnew/v1.0.2.rst [hunk 1]
```diff
@@ -48,6 +48,12 @@ Bug fixes
 
 .. ---------------------------------------------------------------------------
 
+.. _whatsnew_102.privacy.errors:
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.2.rst]: move to 1.1, don't need a separate section, rather you can add the note to Other Enhancements. just say OptionError is now exposed in pandas.errors

**Review by jreback — CHANGES_REQUESTED** [review]
ping on green.

File: doc/source/whatsnew/v1.0.2.rst (hunk 1 above)
Comment (sumanau7) [reply doc/source/whatsnew/v1.0.2.rst]: Done.

**Comment by sumanau7** [comment]
@jreback Please review.

**Review by jorisvandenbossche — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @sumanau7

**Pull request merged**